partner_contact_age_range/models/res_partner_age_range.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2019-Today: Druidoo (<https://www.druidoo.io>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartnerDateRange(models.Model):
+    _name = "res.partner.age.range"
+    _description = "Partner Age Range"
+
+    name = fields.Char(string="Name", required=True)
+    age_from = fields.Integer(string="From", required=True)
```

Comment:
To make sure all age ranges are always covered, you could make age_from a computed field. So you could add Baby with age_to = 2, Toddler with age_to = 4, Child with age_to = 12... etc. Or any other division you want to make. age_from would be the age_to of the range with a lower age_to, or zero if there is no such record. This would also make the check for overlap unneeded, as there can not be any overlap if age_to ust be unique. This would also give the user less work to maintain the ranges.

Reply:
@NL66278 : what will be the value for last range for eg. age for 2 Toddler , age 4 for child , 12 for young is the last range then for young what will be the value for to ?

Reply:
So for Baby the computed age_from would be 0 (as there is no age range with a lesser age_to), for Toddler it would be 3 (the age_to from Baby + 1), for Child it would be 5 (age_to from Toddler +1) etc. For Adult, or for any other oldest age group you want to have, like Pensioner, you could just put a very large age_to, like 128 or if you like 969: https://en.wikipedia.org/wiki/Methuselah

Reply:
@NL66278 : I have add default get to auto set age_from